### PR TITLE
Fix parse string that can contains whitespaces and/or is multiline

### LINKED ISSUE #8239: Mutiline string value in annoation are not parsed correctly
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.18.4            |

Multiline string value in annoation are not parsed correctly.

I have the following annotation
```php
 /**
     * @OA\QueryParameter(name="orderBy", @OA\Schema(type="string"), required=false, description="List of value:
     *  - <b>TRY</b>: To try
     *  - <b>TEST</b>: to test (Default if no parameters given)")
     */
    public function test()
    {}
```
And `description` value is set to `"ListOfValue`
see the following rector diff
```diff
-     * @OA\QueryParameter(name="orderBy", @OA\Schema(type="string"), required=false, description="List of value:
-     *   - <b>TRY</b>: To try
-     *   - <b>TEST</b>: to test (Default if no parameters given)")
+    #[OA\QueryParameter(name: 'orderBy', schema: new OA\Schema(type: 'string'), required: false, description: '"Listofvalue')]
```

## Minimal PHP Code Causing Issue

Will try to make a minimal repo reproduction later.

## Expected Behaviour

`description` parsed value is the same multiline value as the original annoation (minus the `*` and extra space due to annotation)

### PR DESCRIPTION
**Pull request by etshy**
close rectorphp/rector#8239

@samsonasik The fix is kinda dirty for now but made the PR so people can discuss about it.

### CODE REVIEW

File: packages-tests/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser/PlainValueParserTest.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+use PHPStan\PhpDocParser\Lexer\Lexer;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfoFactory;
+use Rector\BetterPhpDocParser\PhpDocInfo\TokenIteratorFactory;
+use Rector\BetterPhpDocParser\PhpDocParser\ClassAnnotationMatcher;
+use Rector\BetterPhpDocParser\PhpDocParser\StaticDoctrineAnnotationParser\PlainValueParser;
+use Rector\Core\PhpParser\Node\BetterNodeFinder;
+use Rector\Testing\PHPUnit\AbstractLazyTestCase;
+use Rector\Testing\TestingParser\TestingParser;
+
+class PlainValueParserTest extends AbstractLazyTestCase
```
Comment (samsonasik) [inline packages-tests/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser/PlainValueParserTest.php:19]: same with https://github.com/rectorphp/rector-src/pull/5123#pullrequestreview-1678697825 , I prefer placing fixture in `tests/Issues` so it is easier what config and rule that causing it.

You can create custom rule under tests/Issues, something like

https://github.com/rectorphp/rector-src/blob/a9908f6498004953531a1a145a4e271fa953a3c7/tests/Issues/PartialValueDocblockUpdate/Source/PartialUpdateTestRector.php#L17

then register in config test 

https://github.com/rectorphp/rector-src/blob/a9908f6498004953531a1a145a4e271fa953a3c7/tests/Issues/PartialValueDocblockUpdate/config/configured_rule.php#L9

along with other config if needed.

**Comment by TomasVotruba** [comment]
I'll check this one...

**Comment by TomasVotruba** [comment]
Thanks for the fix, I cherry picked your commits in https://github.com/rectorphp/rector-src/pull/5317

**Comment by etshy** [comment]
Thanks for continuing this PR and glad my fixes were correct; I wasn't confident changing a bit the parsing process like this at first.
And sorry for not being able to finish this myself.

**Pull request closed without merging**